draft-ietf-httpbis-http2-encryption.md

```diff
@@ -203,81 +200,54 @@ form of protection against active attacks can be provided for clients on subsequ
 
 When an alternative service is able to commit to providing service for a particular origin over TLS
 for a bounded period of time, clients can choose to rely upon its availability, failing when it
-cannot be contacted. Effectively, this makes the choice to use a secured protocol "sticky" in the
-client.
-
-
-## The HTTP-TLS Header Field
+cannot be contacted. Effectively, this makes the choice to use a secured protocol "sticky".
 
-A alternative service can make this commitment by sending a `HTTP-TLS` header field, described here
-using the '#' ABNF extension defined in Section 7 of {{RFC7230}}:
-
-~~~ abnf2616
-    HTTP-TLS     = 1#parameter
-~~~
 
-When it appears in a HTTP response from a strongly authenticated alternative service, this header
-field indicates that the availability of the origin through TLS-protected alternative services is
-"sticky", and that the client MUST NOT fall back to cleartext protocols while this information is
-considered fresh.
+## Opportunistic Commitment
 
-For example:
+A alternative service can commit to providing a secured alternative by including a `commit` member
+in the http-opportunistic well-known resource.  This field includes an RFC3339 [RFC3339] timestamp.
 
 ~~~ example
-    GET /index.html HTTP/1.1
-    Host: example.com
-~~~
-
-~~~ example
-    HTTP/1.1 200 OK
-    Content-Type: text/html
-    Cache-Control: max-age=600
-    Age: 30
-    Date: Thu, 1 May 2014 16:20:09 GMT
-    HTTP-TLS: ma=3600
+{
+  "origins": ["http://example.com:80", "http://www.example.com/:81"],
+  "commit": "2016-03-12T12:27:00Z"
+}
 ~~~
 
-This header field creates a commitment from the origin {{RFC6454}} of the associated resource (in
-the example, `http://example.com`).  For the duration of the commitment, clients SHOULD strongly
-authenticate the server for all subsequent requests made to that origin, though this creates some
-risks for clients (see {{pinrisks}}).
-
-Authentication for HTTP over TLS is described in Section 3.1 of {{RFC2818}}, noting the additional
-requirements in Section 2.1 of {{I-D.ietf-httpbis-alt-svc}}. The header field MUST be ignored if
-strong authentication fails; otherwise, an attacker could create a persistent denial of service by
-falsifying a commitment.
+The value of the `commit` member MUST be ignored unless the alternative service can be strongly
+authenticated.  Minimum authentication requirements for HTTP over TLS are described in Section 2.1
+of {{I-D.ietf-httpbis-alt-svc}} and Section 3.1 of {{RFC2818}}.  As noted in
+{{I-D.ietf-httpbis-alt-svc}}, clients can impose other checks in addition to this minimum set.  For
+instance, a client might choose to apply key pinning {{RFC7469}}.
```

Comment:
So, this is kind of weird, because the .well-known is only for the case when the alternative *isn't* strongly authenticated. Restructuring the spec might help.